arch/cpu/gecko/Makefile.gecko

```diff
@@ -0,0 +1,214 @@
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/pa-conversions
+
+GSDK_C_SRCS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_dma/sl_rail_util_dma.c
+GSDK_SRC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_dma
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_callbacks
+
+GSDK_C_SRCS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_pti/sl_rail_util_pti.c
+GSDK_SRC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_pti
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_pti
+
+GSDK_C_SRCS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_rf_path/sl_rail_util_rf_path.c
+GSDK_SRC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_rf_path
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_rf_path
+
+GSDK_C_SRCS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_rssi/sl_rail_util_rssi.c
+GSDK_SRC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_rssi
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_rssi
+
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/plugin/rail_util_callbacks
+
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/chip/efr32/$(RAIL_FAMILY)
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/common
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/protocol
+GSDK_INC_PATHS += $(GSDK_ROOT)/platform/radio/rail_lib/protocol/ieee802154
+LIBRAIL = $(GSDK_ROOT)/platform/radio/rail_lib/autogen/librail_release/librail_$(RAIL_CPU)_gcc_release.a
+
+CFLAGS += -DSL_COMPONENT_CATALOG_PRESENT
+
+# Contiki
+
+## CPU-dependent directories
+CONTIKI_CPU_DIRS += .
+CONTIKI_CPU_DIRS += sys
+CONTIKI_CPU_DIRS += dev
+CONTIKI_CPU_DIRS += os
+CONTIKI_CPU_DIRS += common
+CONTIKI_CPU_DIRS += storage
+
+## CPU-dependent source files
+
+### dev
+CONTIKI_CPU_SOURCEFILES += gpio-hal-arch.c
+CONTIKI_CPU_SOURCEFILES += uart-arch.c
+CONTIKI_CPU_SOURCEFILES += watchdog-arch.c
+CONTIKI_CPU_SOURCEFILES += efr32-radio.c
+CONTIKI_CPU_SOURCEFILES += efr32-radio-buffer.c
+
+### sys
+CONTIKI_CPU_SOURCEFILES += rtimer-arch.c
+CONTIKI_CPU_SOURCEFILES += clock-arch.c
+CONTIKI_CPU_SOURCEFILES += int-master-arch.c
+CONTIKI_CPU_SOURCEFILES += linkaddr-arch.c
+
+### os
+CONTIKI_CPU_SOURCEFILES += dbg-arch.c
+CONTIKI_CPU_SOURCEFILES += random-arch.c
+
+### storage
+CONTIKI_CPU_SOURCEFILES += cfs-coffee-arch.c
+
+### common
+CONTIKI_CPU_SOURCEFILES += sl_uartdrv_init.c
+
+LDFLAGS += --specs=nosys.specs
+LDFLAGS += -lgcc -lc -lnosys
+LDFLAGS += -Wl,--defsym=_stack=__StackLimit
+LDFLAGS += -Wl,--defsym=_stack_origin=__StackTop
+LDFLAGS += -Wl,--defsym=_heap=__HeapBase
+LDFLAGS += -Wl,--defsym=_eheap=__HeapLimit
+
+CONTIKI_SOURCEFILES += $(CONTIKI_CPU_SOURCEFILES)
+CONTIKI_SOURCEFILES += $(notdir $(GSDK_C_SRCS))
+CONTIKI_SOURCEFILES += $(notdir $(GSDK_ASM_SRCS))
+
+EXTERNALDIRS += $(GSDK_INC_PATHS)
+EXTERNALDIRS += $(GSDK_SRC_PATHS)
+
+TARGET_LIBS = $(LIBRAIL)
```

Comment:
The plan is to migrate Cortex-M from its custom link rule to use the link rule in Makefile.include, so this variable was removed in #2064. Do:
 
```
CONTIKI_OBJECTFILES += $(LIBRAIL)
```
instead, for now.

Reply:
Done